lib/screens/player_screen.dart

```diff
@@ -173,3 +163,107 @@ class _PlayerScreenContent extends StatelessWidget {
     );
   }
 }
+
+enum PlayerHideable {
+  bigPlayButton(14, 14),
+  queueButton(0, 27),
+  progressSlider(0, 14),
+  twoLineTitle(0, 27),
+  codecInfo(0, 20),
+  loopShuffleButtons(96, 0);
+
+  final double width;
+  final double height;
+
+  const PlayerHideable(this.width, this.height);
+}
+
+class PlayerHideableController {
+  final verticalHideOrder = [
+    PlayerHideable.bigPlayButton,
+    PlayerHideable.queueButton,
+    PlayerHideable.codecInfo,
+    PlayerHideable.twoLineTitle,
+    PlayerHideable.progressSlider
+  ];
+
+  List<PlayerHideable> _visible = [];
+  Size? _target;
+
+  void setSizePortrait(Size size) {
+    _visible = List.from(PlayerHideable.values);
+
+    var maxPadding =
+        FinampSettingsHelper.finampSettings.playerScreenCoverMinimumPadding + 8;
+    var targetHeight =
+        size.height + size.width * ((maxPadding / 100.0) * 2 - 1);
+    var targetWidth = size.width;
+    _target = Size(targetWidth, targetHeight);
+
+    if (_getSize().width >= targetWidth) {
+      _visible.remove(PlayerHideable.bigPlayButton);
+      if (_getSize().width >= targetWidth) {
+        _visible.add(PlayerHideable.bigPlayButton);
+        _visible.remove(PlayerHideable.loopShuffleButtons);
+        if (_getSize().width >= targetWidth) {
+          _visible.remove(PlayerHideable.bigPlayButton);
+        }
+      }
+    }
+
+    for (var element in verticalHideOrder) {
+      if (_getSize().height < targetHeight) {
+        return;
+      }
+      _visible.remove(element);
```

Comment:
That might result in many recomputes while rendering, but I'm not sure. Somehow using a local copy and then updating `_visible` might still be a good idea though

Reply:
The hidden widgets get calculated all at once before we continue building the widget tree, so there shouldn't be any recomputes caused by internal operations.